### ISSUE TITLE
Serial generator parameters are not reset to the default values, if a configuration without generator parameters is loaded

### ISSUE THREAD

**Issue by samuelfmlourenco** [bloguetronica/cp2130-conf#55]
This issue can be reproduced by first loading a configuration with generation parameters and then loading another configuration without them. The idea was to allow the user to set those parameters before loading a configuration. However, this behavior is confusing, because settings like "Enable generator by default" and "Auto-generate when loading"  are reset, and the serial generator itself gets disabled, which is as intended, because generator settings shouldn't be preserved.

This behavior is currently implemented in "configurationreader.cpp", inside ConfigurationReader::readFrom(), lines 653 to 655:

```
bool ConfigurationReader::readFrom(QIODevice *device)
{
    serialGeneratorSettings_.doexport = false;  // Default settings if no "generator" element is found
    serialGeneratorSettings_.genenable = false;
    serialGeneratorSettings_.autogen = false;
    ...
}
```

Therefore, a means to reset the serial generator parameters should be implemented.

**Comment by samuelfmlourenco** [comment]
On further analysis, there is no need to add a new function to the SerialGenerator class. Instead, we just need to use the default constructor to set the serial generator parameters to their default values:

```
bool ConfigurationReader::readFrom(QIODevice *device)
{
    serialGeneratorSettings_.serialgen = SerialGenerator();
    serialGeneratorSettings_.doexport = false;
    serialGeneratorSettings_.genenable = false;
    serialGeneratorSettings_.autogen = false;
    ...
}
```

**Comment by samuelfmlourenco** [comment]
Fixed in version 3.1.
